macOS [u]int64 mismatches

In updating Catalyst's embedded Conduit from 0.8.4 to 0.8.6, [this error](https://gitlab.kitware.com/paraview/catalyst/-/jobs/7972350#L290-L296) showed up:

```
/opt/glr/paraview/catalyst-ci/thirdparty/conduit/conduit/conduit_data_type.cpp:39:1: error: static_assert failed due to requirement 'std::is_same<long, long long>::value' "error: conduit_int64 != std::int64_t"
static_assert(std::is_same<conduit_int64, std::int64_t>::value,
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/glr/paraview/catalyst-ci/thirdparty/conduit/conduit/conduit_data_type.cpp:52:1: error: static_assert failed due to requirement 'std::is_same<unsigned long, unsigned long long>::value' "error: conduit_uint64 != std::uint64_t"
static_assert(std::is_same<conduit_uint64, std::uint64_t>::value,
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

I don't see anything interesting in the diff between 0.8.4 and 0.8.6 though…

MR updating Conduit is [here](https://gitlab.kitware.com/paraview/catalyst/-/merge_requests/82/diffs).

--- a/src/libs/conduit/cmake/BitwidthMapping.cmake
+++ b/src/libs/conduit/cmake/BitwidthMapping.cmake
@@ -787,7 +787,6 @@ endmacro()
 # over `long`.
 #-----------------------------------------------------------------------------
 
-set(bitwidth_prefer_long_long FALSE)
 if(CONDUIT_USE_CXX11)
     # note: this call assumes this file included from libs/conduit/CMakeLists.txt
     try_compile( bitwidth_prefer_long_long   # result var
@@ -796,6 +795,8 @@ if(CONDUIT_USE_CXX11)
                  OUTPUT_VARIABLE bitwidth_prefer_long_long_out
                  CXX_STANDARD 11
                  CXX_STANDARD_REQUIRED TRUE)
+else()
+    set(bitwidth_prefer_long_long FALSE)
 endif()
 
 

